bin/openness_report.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+'''
+Usage:
+    openness_report.py --file <od-do-canada.jl.gz> --format <presets.yaml>
+                       --dump <openness_report.csv>
+'''
+import argparse
+import os
+import time
+import datetime
+import sys
+import logging
+import tempfile
+import gzip
+import json
+import yaml
+from collections import defaultdict
+
+from functools import partial
+import traceback
+import unicodecsv
+import codecs
+
+proxy= os.environ['http_proxy']
+
+
+class Records():
+    def __init__(self, file, fmt_file):
+        self.file = file
+        if not os.path.isfile(self.file):
+            self.file = None
+        self.download_file = None
+        with open(fmt_file, 'r') as f:
+            presets = yaml.load(f)
+        fmt = None
+        for it in presets['presets']:
+            if it['preset_name']=='canada_resource_format':
+                fmt = it['values']['choices']
+        if not fmt:
+            raise Exception('no resource format loaded')
+        self.fmt = filter(lambda x: 'openness_score' in x, fmt)
+
+    def __delete__(self):
+        if not self.file:
+            if self.download_file:
+                os.unlink(self.download_file)
+                print('temp file deleted', self.download_file)
+
+    def download(self):
+        if not self.file:
+            # dataset http://open.canada.ca/data/en/dataset/c4c5c7f1-bfa6-4ff6-b4a0-c164cb2060f7
+            url='http://open.canada.ca/static/od-do-canada.jl.gz'
+            r = requests.get(url, stream=True)
+
+            f = tempfile.NamedTemporaryFile(delete=False)
+            for chunk in r.iter_content(1024 * 64):
+                    f.write(chunk)
+            f.close()
```

Comment:
Is there a reason you're downloading the file? you should be able to stream read and yield the records from the request (no temp files to clean up that way)